xbmc/activity/ActivityLog.cpp

```diff
@@ -0,0 +1,298 @@
+/*
+ *      Copyright (C) 2014 Team XBMC
+ *      http://xbmc.org
+ *
+ *  This Program is free software; you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation; either version 2, or (at your option)
+ *  any later version.
+ *
+ *  This Program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with XBMC; see the file COPYING.  If not, see
+ *  <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include "ActivityLog.h"
+#include "FileItem.h"
+#include "GUIUserMessages.h"
+#include "dialogs/GUIDialogKaiToast.h"
+#include "dialogs/GUIDialogSelect.h"
+#include "filesystem/ActivitiesDirectory.h"
+#include "guilib/GUIWindowManager.h"
+#include "profiles/ProfilesManager.h"
+#include "settings/Settings.h"
+#include "threads/SingleLock.h"
+#include "utils/Variant.h"
+
+CActivityLog::CActivityLog()
+  : m_activities(),
+    m_activitiesMap()
+{ }
+
+CActivityLog::~CActivityLog()
```

Comment:
.. and you'll get a virtual method but no virtual destructor warning (which is spot on).